src/LineBot/Actions/DateTimePickerAction.cs

```diff
@@ -0,0 +1,195 @@
+        [JsonConverter(typeof(EnumConverter<ActionType>))]
+        private readonly ActionType _type = ActionType.DateTimePicker;
+#pragma warning restore 0414
+
+        private string _label;
+        private string _data;
+        private DateTimePickerMode _mode;
+        private DateTime? _initial;
+        private DateTime? _min;
+        private DateTime? _max;
+
+        /// <summary>
+        /// Gets or sets the label.
+        /// <para>Max: 20 characters.</para>
+        /// </summary>
+        [JsonProperty("label")]
+        public string Label
+        {
+            get
+            {
+                return _label;
+            }
+
+            set
+            {
+                if (string.IsNullOrWhiteSpace(value))
+                    throw new InvalidOperationException("The label cannot be null or whitespace.");
+
+                if (value.Length > 20)
+                    throw new InvalidOperationException("The label cannot be longer than 20 characters.");
+
+                _label = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the string returned via webhook in the postback.data property of the <see cref="IPostback"/> event.
+        /// <para>Max: 300 characters.</para>
+        /// </summary>
+        [JsonProperty("data")]
+        public string Data
+        {
+            get
+            {
+                return _data;
+            }
+
+            set
+            {
+                if (string.IsNullOrWhiteSpace(value))
+                    throw new InvalidOperationException("The data cannot be null or whitespace.");
+
+                if (value.Length > 300)
+                    throw new InvalidOperationException("The data cannot be longer than 300 characters.");
+
+                _data = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the action mode for the datetimepicker.
+        /// </summary>
+        [JsonProperty("mode")]
+        [JsonConverter(typeof(EnumConverter<DateTimePickerMode>))]
+        public DateTimePickerMode Mode
+        {
+            get
+            {
+                return _mode;
+            }
+
+            set
+            {
+                _mode = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the initial datetime for the datetimepicker.
+        /// </summary>
+        [JsonProperty("initial")]
+        public DateTime? Initial
+        {
+            get
+            {
+                return _initial;
+            }
+
+            set
+            {
+                _initial = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the min datetime for the datetimepicker.
+        /// </summary>
+        [JsonProperty("min")]
+        public DateTime? Min
+        {
+            get
+            {
+                return _min;
+            }
+
+            set
+            {
+                if (Max.HasValue && value > Max.Value)
```

Comment:
I think this check needs to be changed. The minimal difference should be at least one minute or a day depending on the mode? We should probably move the mode to the constructor and make it read only so it will be easier to add checks for that.